sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -79,18 +81,17 @@ internal HttpMessage CreateUploadRequest(string ruleId, string streamName, Reque
         /// <summary>
         /// Hidden method for batching data - serializing into arrays of JSON no more than SingleUploadThreshold each
         /// </summary>
-        /// <typeparam name="T"></typeparam>
         /// <param name="logEntries"></param>
         /// <param name="options"></param>
         /// <returns></returns>
-        internal static IEnumerable<BatchedLogs<T>> Batch<T>(IEnumerable<T> logEntries, UploadLogsOptions options = null)
+        internal static IEnumerable<BatchedLogs> Batch(IEnumerable<object> logEntries, UploadLogsOptions options = null)
```

Comment:
Can we verify that the change from T to object is necessary?

Reply:
I think this is because of the use of SyncAsyncEventArgs and RaiseAsync? @tg-msft

Reply:
No, we shouldn't do this here.  We should leave this alone but cast `T` to `object` when passing entries into the event args/helpers so we don't need to make all of them strongly typed (i.e., we don't want to make it `UploadLogsOptions<T>` to have a `SyncAsyncEventHandler<UploadFailedEventArgs<T>>`, etc., so we'll just use `object` there and people can cast back to `T` if they want).